src/txn/txn_timestamp.c

```diff
@@ -705,6 +738,28 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
 	return (0);
 }
 
+/*
+ * __wt_txn_query_timestamp --
+ *	Query a timestamp.
+ */
+int
+__wt_txn_query_timestamp(
```

Comment:
It's a bit sad that so much of this is duplicated with the existing `__wt_txn_global_query_timestamp` - did you think about instead renaming that to `__wt_txn_query_timestamp` and adding a bool parameter to differentiate between the session and global versions?